### DISCUSSION TITLE
Can wire support running a function after a component is created?

### DISCUSSION THREAD

**Discussion by Lunaticf** [google/wire#276] [Q&A]
like a PostConstruct in java?
Assume a mq, start it after mq component created

**Comment by zombiezen** [comment]
You could create a wrapper function that calls the generated injector function and then run code after calling the generated function. Is that what you mean?

**Reply by Lunaticf** [reply]
Thx! you mean create a Newxxx function，with the injected dependencies, that is ok~
just not the original responsibility of Newxxx to build some component, instead of running some code

**Reply by zombiezen** [reply]
Oh I see, you're talking about the `@PostConstruct` annotation in Spring. There isn't exactly a direct analogue in Wire, since there isn't the same concept of constructor in Go. You should make your provider function (the Newxxx function) call whatever functions or methods it needs to before returning the component.
